lib/situation_room/site.ex

```diff
@@ -1,16 +1,88 @@
 defmodule SituationRoom.Site do
+  @moduledoc """
+  Documentation for `SituationRoom.Site`.
+  """
   use Ecto.Schema
   import Ecto.Changeset
+  import SituationRoom.Repo
 
   schema "sites" do
-    field :endpoint, :string
-    field :name, :string
+    field(:endpoint, :string)
+    field(:name, :string)
   end
 
+  # Delete a site from the database by specifying specific field
+  # param ex: (name: "mojotech") or (endpoint: "http://mojo.com")
+  def delete_site(param) do
+    case res = get_site(param) do
+      %SituationRoom.Site{} ->
+        delete(res)
+
+      _ ->
+        {:error, "Site does not exist"}
+    end
+
+    # Rescues when there is an unexpected error.. eg. duplicate entries found in db
+  rescue
+    _ -> {:error, "Cannot delete site"}
+  end
+
+  # Get a site from the database by one specific field
+  # param ex: (name: "mojotech") or (endpoint: "http://mojo.com")
+  def get_site(param) do
+    SituationRoom.Site |> get_by(param)
```

Comment:
We don't typically use pipes that only have a single "link" in the chain.  Just call the function directly.

    get_by(SituationRoom.Site, param)

Same thing below on line 38

    all(SituationRoom.Site)